python/ray/autoscaler/_private/monitor.py

```diff
@@ -177,7 +228,29 @@ def update_load_metrics(self):
             request, timeout=4)
         resources_batch_data = response.resource_usage_data
 
+        # Tell the readonly node provider what nodes to report.
+        if self.readonly_config:
+            self.autoscaler.provider._set_last_batch(
+                list(resources_batch_data.batch))
+
+        mirror_node_types = {}
+        resource_deadlock = False
         for resource_message in resources_batch_data.batch:
+            # Generate node type config based on GCS reported node list.
+            if self.readonly_config:
+                # Keep prefix in sync with ReadonlyNodeProvider.
+                node_type = "node_{}".format(resource_message.node_id.hex())
+                if resource_message.resource_deadlock_detected:
```

Comment:
If we check `hasattr(resource_message, resource_deadlock_detected)` would that be enough to preserve compatibility of new autoscaler with old gcs?